### ISSUE TITLE
ESBMC fails to reason about array with nondet size

### ISSUE THREAD

**Issue by feliperodri** [esbmc/esbmc#28]
Consider the following example:

```
struct aws_string {
    const unsigned long len;
    const unsigned char bytes[];
};

void *bounded_malloc(unsigned long size) {
    __ESBMC_assume(size <= ((18446744073709551615UL) >> (8 + 1)));
    return malloc(size);
}

static inline const unsigned char *aws_string_bytes(const struct aws_string *str) {
    return str->bytes;
}

int main() {
    unsigned long len;
    __ESBMC_assume(len < ((18446744073709551615UL) - 1 - sizeof(struct aws_string)));

    struct aws_string *str = bounded_malloc(sizeof(struct aws_string) + len + 1);
    *(unsigned long *)(&str->len) = len;
    *(unsigned char *)&str->bytes[len] = '\0';

    __ESBMC_assert(aws_string_bytes(str) == str->bytes, "This should never fail");

    __ESBMC_assert(((((str->len + 1)) == 0) || ((&str->bytes[0]))), "This should never fail");
    __ESBMC_assert(str->bytes[str->len] == 0, "This should never fail");
}
```

ESBMC fails the assert statements.

Tested with the following command:
```
esbmc --no-div-by-zero-check --force-malloc-success --state-hashing --no-align-check --k-step 2 --floatbv --unlimited-k-steps --context-bound 2 main.c --64 --k-induction --max-inductive-step 3 --no-pointer-check --no-bounds-check
```

**Comment by rafaelsamenezes** [comment]
Closed by https://github.com/esbmc/esbmc/pull/127

**Comment by rafaelsamenezes** [comment]
PR https://github.com/esbmc/esbmc/pull/127 solves this?

**Comment by mikhailramalho** [comment]
Nope, we need to implement the support for flexible arrays in the pointer analysis.

**Comment by lucasccordeiro** [comment]
@rafaelsamenezes: What is the current status of the flexible array implementation?

**Comment by lucasccordeiro** [comment]
Closed via bf175c0611d321fcbaf2c2eb206d20062a7b5bb0.
